x-pack/plugins/cases/server/plugin.ts

```diff
@@ -144,8 +150,25 @@ export class CasePlugin {
         logger: this.log,
       });
     };
+    const getCasesClient = (
```

Comment:
I might be missing something but could we use `getCasesClientWithRequestAndContext` instead of creating a new one here? It looks like in the security solution code we have the `context` and `request` right?

Reply:
right, that seemed like it was the intentional path. But the `context` it expects is a `CasesRequestHandlerContext` which is this:

```js
{
    cases: interface  CaseRequestContext { 
        getCasesClient(): CasesClient
    }
    actions: interface ActionsApiRequestHandlerContext {
        getActionsClient: () => ActionsClient;
        listTypes: ActionTypeRegistry['list'];
    }
}
```

Creating this context which has something that provides a CasesClient seemed a bit circular, and not something I thought was easy to construct from the outside perhaps. So that's why I just added another entry point to just get this working

Reply:
oooh, gotcha. Makes sense. Yeah you can leave it as it is or switch the `context` type to be `RequestHandlerContext` that should work too, I don't think that function actually needs the `CasesRequestHandlerContext`. Either is fine, we actually change that function to not need the context in our feature branch.

Reply:
good call. You're right, nothing was depending on that being a CasesReqContext, so setting as a standard ReqContext didn't break anything. So now I can just call that. 

now this PR does very little to Cases. It exposes that `getCasesByAlertIds`, and the rest is just adding an interface type to reference. :tada: